src/NUnitEngine/nunit.engine/IDriverService.cs

```diff
@@ -37,7 +37,8 @@ public interface IDriverService
         /// </summary>
         /// <param name="domain">The AppDomain in which to run the tests</param>
         /// <param name="assemblyPath">The path to the test assembly</param>
+        /// <param name="targetFramework">The value of any TargetFrameworkAttribute on the assembly, or null</param>
         /// <returns></returns>
-        IFrameworkDriver GetDriver(AppDomain domain, string assemblyPath);
+        IFrameworkDriver GetDriver(AppDomain domain, string assemblyPath, string targetFramework);
```

Comment:
This interface is internal to the engine, so changing it has no compatibility impact.